index.bs

```diff
@@ -1386,6 +1400,18 @@ Each {{XRView}} has an associated <dfn for="XRView">internal projection matrix</
 
 Note: The {{XRView/transform}} can be used to position camera objects in many rendering libraries. If a more traditional view matrix is needed by the application one can be retrieved by calling `view.transform.inverse.matrix`.
 
+<div class="algorithm" data-algorithm="request-viewport-scale">
+
+The <dfn method for="XRView">requestViewportScale(|scale|)</dfn> method requests that the user agent should set the [=viewport scale=] for this viewport to the requested value.
+
+When this method is invoked on a [=view=] |view|, the user agent MUST run the following steps:
+
+  1. If the [=view/viewport modifiable=] flag is `false` abort these steps.
+  1. If |scale| is greater than 1.0, set |scale| to 1.0.
+  1. If |scale| is smaller than |view|'s [=view/minimum viewport scale=], set it to the [=view/minimum viewport scale=].
+  1. Set the |view|'s [=view/viewport scale=] value to |scale|.
+
+</div>
```

Comment:
I appreciate the idea that some non-primary views may want to scale separately from the primary views, and though it's a little weird to me that the primary views could scale their viewports separately it's something that appears to be well supported in the various backends.

Let's theorize about a backend that must have some/all of it's views using the same viewport size for whatever reason, though. The only compatibility path I see for that scenario is to have it force the `minimum viewport scale` to `1.0` across the board, thus effectively disabling this feature. (Though frankly if your backend is limited like that maybe it couldn't have supported this feature anyway?)

Does anyone think that's not sufficient? Any we might want a non-normative note that says something about how that case can be handled.

Reply:
I don't have a strong opinion on this one, but IIRC there was a desire to support per-viewport scaling. I think the suggested workaround of disabling viewport scaling in the unlikely case that it's not supportable on a platform seems acceptable, but please speak up if this is concerning.